src/main/java/com/plugish/woominecraft/Util/RcHttp.java

```diff
@@ -122,15 +117,23 @@ public String send( String url, HashMap<String, String> hashMap ) throws Excepti
 		}
 
 		// Some debug logging.
-		if ( plugin.getConfig().getBoolean( "debug", false ) ) {
-			plugin.wmc_log( "Sending Request" );
-			plugin.wmc_log( "URL Config Field: " + plugin.getConfig().getString( "url", "empty" ) );
-			plugin.wmc_log( "Headers: " + Arrays.toString( response.getAllHeaders() ) );
-			plugin.wmc_log( "HTTP Response Code: " + response.getStatusLine().getStatusCode() );
-			plugin.wmc_log( "Content Body Snippet (128 chars): " + resultString.substring( 0, Length ) );
+		if ( getPlugin().getConfig().getBoolean( "debug", false ) ) {
+			debug();
 		}
 
 		client.close();
 		return resultString;
 	}
+
+	private void debug() {
+		getPlugin().wmc_log( "Sending Request" );
+		getPlugin().wmc_log( "URL Config Field: " + getPlugin().getConfig().getString( "url", "empty" ) );
+		getPlugin().wmc_log( "Headers: " + Arrays.toString( response.getAllHeaders() ) );
+		getPlugin().wmc_log( "HTTP Response Code: " + response.getStatusLine().getStatusCode() );

```

Comment:
Apache httpClient is not `thread-safe` you may not get the proper status code here.